modules/core/04-channel/keeper/packet.go

```diff
@@ -467,6 +467,11 @@ func (k Keeper) AcknowledgePacket(
 	// Delete packet commitment, since the packet has been acknowledged, the commitement is no longer necessary
 	k.deletePacketCommitment(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
 
+	if channel.FlushStatus == types.FLUSHING && !k.hasInflightPackets(ctx, packet.GetSourcePort(), packet.GetSourceChannel()) {
```

Comment:
could test this using a separate function to examine state post-AcknowledgePacket. Alternatively, as we slightly discussed with Cian, tests for packet handlers could be added in the upgrade tests. Lmk what people think.

Reply:
I'm okay with this for now. We can maybe add some upgrade tests later to make stronger assumptions